util/src/main/scala/geotrellis/util/StreamingByteReader.scala

```diff
@@ -112,13 +118,14 @@ class StreamingByteReader(rangeReader: RangeReader, chunkSize: Int = 45876) exte
 
     if (filePosition != chunkRange.start + chunkBuffer.position)
       chunkBuffer.position((filePosition - chunkRange.start).toInt)
+    trimmed.toInt
   }
 
   def getBytes(length: Int): Array[Byte] = {
-    ensureChunk(length)
-    val bytes = Array.ofDim[Byte](length)
+    val ensuredLength = ensureChunk(length)
+    val bytes = Array.ofDim[Byte](ensuredLength)
     chunkBuffer.get(bytes)
-    filePosition += length
+    filePosition += ensuredLength
```

Comment:
This set of changes was necessary to get tests to pass. `getBytes` now only attempts to read the length of bytes which can be ensured by `ensureChunk`